File: iosApp/iosApp/Views/FeaturedArticleCard.swift
import SwiftUI
import shared

struct FeaturedArticleCard: View {
    let headline: Headline
    var onClick: () -> Void
    
    init(headline: Headline, onClick: @escaping () -> Void = {}) {
        self.headline = headline
        self.onClick = onClick
    }
    
    var body: some View {
        Button(action: onClick) {
            ZStack {
                // Background image
                AsyncImage(url: URL(string: headline.imageUrl ?? "")) { image in
                    image
                        .resizable()
                        .aspectRatio(contentMode: .fill)
                } placeholder: {
                    Rectangle()
                        .fill(Color.gray.opacity(0.3))
                        .overlay(
                            Image(systemName: "photo")
                                .foregroundColor(.gray)
                        )
                }
                .frame(width: 250, height: 200)
                .clipped()
                
                // Dark overlay gradient
                LinearGradient(
                    gradient: Gradient(colors: [Color.clear, Color.black.opacity(0.7)]),
                    startPoint: .top,
                    endPoint: .bottom
                )
                
                // Content overlay
                VStack(alignment: .leading, spacing: 8) {
                    Spacer()
                    
                    Text(headline.title)
                        .font(.headline)
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                        .lineLimit(2)
                        .multilineTextAlignment(.leading)
                    
                    HStack {
                        HStack(spacing: 8) {
                            // Author avatar placeholder
                            Circle()
                                .fill(Color.gray.opacity(0.6))
                                .frame(width: 24, height: 24)
                            
                            VStack(alignment: .leading, spacing: 2) {
                                Text(headline.source)
                                    .font(.caption)
                                    .fontWeight(.medium)
                                    .foregroundColor(.white)
                                
                                Text(formatDate(headline.publishedAt))
                                    .font(.caption2)
                                    .foregroundColor(.white.opacity(0.8))
                            }
                        }
                        
                        Spacer()
                        
                        Button(action: {
                            // Share action
                        }) {
                            Image(systemName: "square.and.arrow.up")
                                .foregroundColor(.white)
                                .frame(width: 20, height: 20)
                        }
                        .frame(width: 32, height: 32)
                    }
                }
                .padding(16)
            }
        }
        .frame(width: 250, height: 200)
        .cornerRadius(16)
        .shadow(radius: 4)
        .buttonStyle(PlainButtonStyle())
    }
    
    private func formatDate(_ dateString: String) -> String {
        // Simple date formatting - in a real app, you'd use proper date formatting
        return "Sep 9, 2022"
    }
}

struct FeaturedArticleCard_Previews: PreviewProvider {
    static var previews: some View {
        FeaturedArticleCard(headline: Headline(
            id: "1",
            title: "Sample Article Title",
            description: "Sample description",
            url: "https://example.com",
            imageUrl: "https://example.com/image.jpg",
            publishedAt: "2022-09-09T12:00:00Z",
            source: "Sample Source"
        ))
    }
}

File: androidApp/src/main/java/com/kashif/kmmnewsapp/android/components/FeaturedArticleCard.kt
package com.kashif.kmmnewsapp.android.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Share
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.draw.clip
import coil3.compose.AsyncImage
import com.kashif.kmmnewsapp.android.R
import com.kashif.kmmnewsapp.feature.headlines.domain.Headline

@Composable
fun FeatureCard(
    headline: Headline,
    modifier: Modifier = Modifier,
    onClick: () -> Unit = {}
) {
    Card(
        modifier = modifier
            .width(250.dp)
            .height(200.dp)
            .clickable { onClick() },
        shape = RoundedCornerShape(16.dp),
        elevation = CardDefaults.cardElevation(defaultElevation = 4.dp)
    ) {
        Box {
            AsyncImage(
                model = headline.imageUrl,
                contentDescription = headline.title,
                contentScale = ContentScale.Crop,
                placeholder = painterResource(R.drawable.ic_placeholder),
                error = painterResource(R.drawable.ic_error),
                modifier = Modifier.fillMaxSize()
            )
            
            // Dark overlay for text readability
            Box(
                modifier = Modifier
                    .fillMaxSize()
                    .background(
                        Brush.verticalGradient(
                            colors = listOf(
                                Color.Transparent,
                                Color.Black.copy(alpha = 0.7f)
                            )
                        )
                    )
            )
            
            // Article info overlay
            Column(
                modifier = Modifier
                    .align(Alignment.BottomStart)
                    .padding(16.dp)
            ) {
                Text(
                    text = headline.title,
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                    color = Color.White,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis
                )
                
                Spacer(modifier = Modifier.height(8.dp))
                
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        // Author avatar placeholder
                        Box(
                            modifier = Modifier
                                .size(24.dp)
                                .background(
                                    Color.Gray.copy(alpha = 0.6f),
                                    shape = CircleShape
                                )
                        )
                        
                        Spacer(modifier = Modifier.width(8.dp))
                        
                        Column {
                            Text(
                                text = headline.source,
                                style = MaterialTheme.typography.labelMedium,
                                color = Color.White,
                                fontWeight = FontWeight.Medium
                            )
                            Text(
                                text = formatDate(headline.publishedAt),
                                style = MaterialTheme.typography.labelSmall,
                                color = Color.White.copy(alpha = 0.8f)
                            )
                        }
                    }
                    
                    // Share icon
                    IconButton(
                        onClick = { /* TODO: Implement share */ },
                        modifier = Modifier.size(32.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Share,
                            contentDescription = "Share",
                            tint = Color.White,
                            modifier = Modifier.size(20.dp)
                        )
                    }
                }
            }
        }
    }
}

private fun formatDate(dateString: String): String {
    // Simple date formatting - in a real app, you'd use proper date formatting
    return "Sep 9, 2022"
}